instrumentation/jms/src/main/java/brave/jms/PropertyFilter.java

```diff
@@ -40,17 +43,20 @@ static void filterProperties(Message message, Set<String> namesToClear) {
   }
 
   static void filterProperties(Message message, Set<String> namesToClear, List<Object> out) {
-    Enumeration<?> names;
+    List<String> names;
     try {
-      names = message.getPropertyNames();
+      names = list(message.getPropertyNames());
     } catch (Throwable t) {
       propagateIfFatal(t);
       log(t, "error getting property names from {0}", message, null);
       return;
     }
+    
+    if (disjoint(namesToClear, names) && message instanceof BytesMessage) {
```

Comment:
This is maybe a naive comment but could we a dd a comment on the reasoning for `message instanceof BytesMessage` ?

Reply:
This line is to avoid unnecessary clearing properties when there's nothing to clear. I would like to remove the condition. Could you please help me fix https://github.com/openzipkin/brave/blob/master/instrumentation/jms/src/test/java/brave/jms/ITJms_1_1_TracingMessageConsumer.java 

It fails on line 300 receive_customSampler.

Reply:
I don't think I have the right knowledge to help you on this matter. @jeqo or @anuraaga might help. My comment is more for keeping it readable even for someone like me who has very little context.

Reply:
@me3x if you check `disjoint` implementation is actually doing the same as we are doing on the loop later:

```
for (Object e : iterate) {
            if (contains.contains(e)) {
               // Found a common element. Collections are not disjoint.
                return false;
            }
        }
```

I'd like to propose to refactor this logic. Here is one way: we can remove this check and go directly to the `for` loop. Loop logic can be restructured to first do `if (!namesToClear.contains(name))` and if true, then extract value and check if not null, add to `out`.

Reply:
I refactored as you proposed. Please check.